sale_order_invoicing_queued/models/sale_order.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+try:
+    from odoo.addons.queue_job.job import job
+except ImportError:
+    import logging
+    import functools
+
+    logging.getLogger().debug("Can't `import queue_job`.")
+
+    def empty_decorator_factory(*argv, **kwargs):
+        return functools.partial
+    job = empty_decorator_factory
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    invoicing_job_ids = fields.Many2many(
```

Comment:
:thinking: By using a m2m, do you mean that there could be several jobs generating the same invoice for the same SO?

Reply:
Not the same invoice, but several invoices. One sales order > several invoices, thus several jobs.